File: TeamCode/src/main/java/org/riverdell/robotics/autonomous/detection/SampleType.java
package org.riverdell.robotics.autonomous.detection;

import org.opencv.core.Scalar;

public enum SampleType {
    BLUE(
        new Scalar(90.0, 100.0, 100.0),
        new Scalar(130.0, 255.0, 255.0)
    ),
    YELLOW(
        new Scalar(20.0, 100.0, 100.0),
        new Scalar(30.0, 255.0, 255.0)
    ),
    RED(
        new Scalar(0.0, 100.0, 100.0),
        new Scalar(10.0, 255.0, 255.0)
    );

    private final Scalar colorRangeMinimum;
    private final Scalar colorRangeMaximum;

    SampleType(Scalar colorRangeMinimum, Scalar colorRangeMaximum) {
        this.colorRangeMinimum = colorRangeMinimum;
        this.colorRangeMaximum = colorRangeMaximum;
    }

    public Scalar getColorRangeMinimum() {
        return colorRangeMinimum;
    }

    public Scalar getColorRangeMaximum() {
        return colorRangeMaximum;
    }
}

File: TeamCode/src/main/java/org/riverdell/robotics/autonomous/detection/SampleType.kt
package org.riverdell.robotics.autonomous.detection

import org.opencv.core.Scalar

enum class SampleType(
    val colorRangeMinimum: Scalar,
    val colorRangeMaximum: Scalar
) {
    Blue(
        colorRangeMinimum = Scalar(90.0, 100.0, 100.0),
        colorRangeMaximum = Scalar(130.0, 255.0, 255.0)
    ),
    Yellow(
        colorRangeMinimum = Scalar(20.0, 100.0, 100.0),
        colorRangeMaximum = Scalar(30.0, 255.0, 255.0)
    ),
    Red(
        colorRangeMinimum = Scalar(0.0, 100.0, 100.0),
        colorRangeMaximum = Scalar(10.0, 255.0, 255.0)
    )
}
